### PR TITLE
first step record playback test

### PR DESCRIPTION
**Pull request by vhvb1989**
Following JAVA design.
Adding basic required classes for the test record-playback-live infrastructure.

Only live mode for now. I want to first share how this would look like and used.
Then I can continue with more features for this. 
I can keep this branch and do new PRs to this one if we want to wait until this is finished to merge to master

Next steps (not covered here yet) :
TestResourceNam (for uuid, secrets, etc)
Recording implementation
Playback implementation

### CODE REVIEW

**Comment by Jinming-Hu** [comment]
how are you going to handle

1. blob name, blob content, blob length etc. that randomly generated
2. multi thread uploading/downloading. The order of those requests from multiple threads will be undetermined

**Comment by vhvb1989** [comment]
> how are you going to handle
> 
> 1. blob name, blob content, blob length etc. that're randomly generated
> 2. multi thread uploading/downloading. The order of those requests from multiple threads will be undetermined

Those things were already solved for the other languages, I am just adding the C++ equivalent.
I know there are some utilities I need to still create to handle the random stuff and replacing/hiding sensitive data. This is just the first part of more things still required.

About the multithread, each request for each thread is different than the others, isn't it?  they either set a different value for a header value or query parameter to get a different result from the server. 
The requests are all recorded with all this properties, and the playback policy will look a request that matches that request to use the recorded data.  If the data from the request is a random value, I believe it is also handled.  

I will explain once I get to that point :)

File: sdk/core/azure-core-test/cgmanifest.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+                "Other": {
+                    "Name": "clang-format",
+                    "Version": "9.0.0-2",
+                    "DownloadUrl": "https://ubuntu.pkgs.org/18.04/ubuntu-updates-universe-amd64/clang-format-9_9-2~ubuntu18.04.2_amd64.deb.html"
+                }
+            },
+            "DevelopmentDependency": true
+        }
+    ]
+}
```
Comment (vhvb1989) [inline sdk/core/azure-core-test/cgmanifest.json:25]:
```suggestion
}

```

**Comment by vhvb1989** [comment]
Moving branch to this repo so I can create PRs from my fork to this branch directly instead of creating PRs in my fork only

**Pull request closed without merging**